editor/components/meta-boxes/meta-boxes-area/index.js

```diff
@@ -1,153 +1,85 @@
+	 */
 	componentDidMount() {
-		this.mounted = true;
-		this.fetchMetaboxes();
+		this.form = document.querySelector( '.metabox-location-' + this.props.location );
+		this.node.appendChild( this.form );
 	}
 
+	/**
+	 * Get the meta box location form from the original location.
+	 */
 	componentWillUnmount() {
-		this.mounted = false;
-		this.unbindFormEvents();
 		document.querySelector( '#metaboxes' ).appendChild( this.form );
 	}
 
-	unbindFormEvents() {
-		if ( this.form ) {
-			this.form.removeEventListener( 'change', this.checkState );
-			this.form.removeEventListener( 'input', this.checkState );
-		}
-	}
-
-	componentWillReceiveProps( nextProps ) {
-		if ( nextProps.isUpdating && ! this.props.isUpdating ) {
-			this.setState( { loading: true } );
-			const { location } = nextProps;
-			const headers = new window.Headers();
-			const fetchOptions = {
-				method: 'POST',
-				headers,
-				body: new window.FormData( this.form ),
-				credentials: 'include',
-			};
-
-			// Save the metaboxes
-			window.fetch( addQueryArgs( window._wpMetaBoxUrl, { meta_box: location } ), fetchOptions )
-				.then( () => {
-					if ( ! this.mounted ) {
-						return false;
-					}
-					this.setState( { loading: false } );
-					this.props.metaBoxReloaded( location );
-				} );
-		}
-	}
-
-	fetchMetaboxes() {
-		const { location } = this.props;
-		this.form = document.querySelector( '.metabox-location-' + location );
-		this.node.appendChild( this.form );
-		this.form.onSubmit = ( event ) => event.preventDefault();
-		this.originalFormData = this.getFormData();
-		this.form.addEventListener( 'change', this.checkState );
-		this.form.addEventListener( 'input', this.checkState );
-		this.props.metaBoxLoaded( location );
-	}
-
-	getFormData() {
-		return jQuery( this.form ).serialize();
-	}
-
-	checkState() {
-		const { loading } = this.state;
-		const { isDirty, changedMetaBoxState, location } = this.props;
-
-		const newIsDirty = ! isEqual( this.originalFormData, this.getFormData() );
-
-		/**
-		 * If we are not updating, then if dirty and equal to original, then set not dirty.
-		 * If we are not updating, then if not dirty and not equal to original, set as dirty.
-		 */
-		if ( ! loading && isDirty !== newIsDirty ) {
-			changedMetaBoxState( location, newIsDirty );
-		}
+	/**
+	 * Binds the metabox area container node.
+	 *
+	 * @param {Element} node DOM Node.
+	 */
+	bindNode( node ) {
+		this.node = node;
 	}
 
+	/**
+	 * @inheritdoc
+	 */
 	render() {
-		const { location } = this.props;
-		const { loading } = this.state;
+		const { location, isSaving } = this.props;
 
 		const classes = classnames(
 			'editor-meta-boxes-area',
 			`is-${ location }`,
 			{
-				'is-loading': loading,
+				'is-loading': isSaving,
 			}
 		);
 
 		return (
 			<div className={ classes }>
-				{ loading && <Spinner /> }
-				<div ref={ this.bindNode } />
+				{ isSaving && <Spinner /> }
+				<div className="editor-meta-boxes-area__container" ref={ this.bindNode } />
```

Comment:
Unrelated: When I see `bindNode`, I assume it's assigning the `ref` of the root return value, not a child of the element. A minor thing, but I think it can be nice to clarify in the function name, e.g. `bindContainerNode`.